app/src/org/commcare/activities/connect/ConnectActivity.java

```diff
@@ -69,22 +89,82 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
                     .setPopUpTo(navController.getGraph().getStartDestinationId(), true)
                     .build();
             navController.navigate(fragmentId, bundle, options);
+        } else {
+            if (redirectionAction != null) {
+                ConnectManager.init(this);
+                ConnectManager.unlockConnect(this, success -> {
+                    if (success) {
+                        getJobDetails();
+                    }
+                });
+            }
+        }
+    }
+
+    /**
+     * Returns the fragment ID based on the redirection action.
+     * <p>
+     * This method determines which fragment should be displayed based on the value of the redirectionAction.
+     * It maps specific actions to their corresponding fragment IDs.
+     *
+     * @return The ID of the fragment to be displayed.
+     */
+    private int getFragmentId() {
+        int fragmentId;
+        if (redirectionAction.equals("ccc_opportunity_summary_page")) {
```

Comment:
Let's make these strings private static final strings defined for the class (so we only have to type them out once).

Reply:
Done